### ISSUE TITLE
Cannot use Lava variable as array index after math filters

### ISSUE THREAD

**Issue by cabal95** [SparkDevNetwork/Rock#1943]
### Prerequisites

* [X] Can you reproduce the problem on a fresh install or the [demo site](http://rock.rocksolidchurchdemo.com/)?
* [X] Did you include your Rock version number and [client culture](https://github.com/SparkDevNetwork/Rock/wiki/Environment-and-Diagnostics-Information) setting?
* [X] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ASparkDevNetwork%2FSlack) to see if your bug or enhancement is already reported?

### Description

When assigning a number to a Lava variable and then using one of the math filters (Plus, Minus, etc.), the variable somehow becomes non-numeric (maybe non-integer only?) and can no longer be used as an index variable into an array.

### Steps to Reproduce

Consider the following Lava run through the Lava Tester:

```
{% assign array = "value1,value2,value3,value4" | Split:"," %}
{% assign ind = 0 %}
{% assign ind = ind | Plus:1 %}

{% for item in array %}
  {% if forloop.index == ind %}{% assign index = forloop.index %}{% endif %}
{% endfor %}

Fails: {{ array[ind] }} ({{ind}})<br />
Works: {{ array[index] }} ({{index}})
```

The first chunk creates an array and sets up the index variable, initially to 0 and then adds 1. The second chunk uses a forloop to setup another variable to be the same value but forced as an integer. The third block displays the results.

> Fails: (1)
> Works: value2 (1)

We tried a few workarounds like passing the variable through the Floor filter but we were never able to get the value back into a pure-integer form that the [] accessor would allow. I can't tell if this is a bug in the filters somehow converting the value into a string (even though they return as a decimal); or because they are returning as a decimal the [] accessor does not recognize them as a valid index number.

**Expected behavior:** 

Index access should work anytime a variable contains a whole number.

**Actual behavior:** 

Empty string is returned, as if an invalid index was specified.

### Versions

* **Rock Version:** 1.5.4 &   
* **Client Culture Setting** en-US

**Comment by edmistj** [comment]
@cabal95 Humm those filters are actually in the base dotLiquid library (https://github.com/dotliquid/dotliquid/blob/e9a8d3aba9cd88bd1ce9f16908dfedf27b1551ec/src/DotLiquid/StandardFilters.cs#L559) 

Looks like they return as objects. I'm guessing since they could be integers, decimals, etc. Would perhaps having a ToInteger Lava filter help?

**Comment by cabal95** [comment]
Possibly. I don't have a full Rock dev environment to test custom Rock.dll code, but it was something we discussed in Slack. It depends on the issue being it returning an object/decimal OR if something higher up the call stack is converting the value returned to a string no matter what.

Another solution would be an "Index" filter as we could also simplify things, e.g.
```
{{ Workflow | Attribute:'MyCSVAttribute' | Split:',' | Index:2 }}
```

Either would solve the problem.

**Comment by edmistj** [comment]
I like the Index idea. I'll add that to the ToDo.

**Comment by cabal95** [comment]
Fixed via d65369c9bb0c38891448a0c53296271b34eaddfa
